authentication/serializers.py

```diff
@@ -1,66 +1,81 @@
 from django.contrib.auth import authenticate, get_user_model
 from django.conf import settings
 from django.core.exceptions import ValidationError
+from rest_framework.validators import UniqueValidator
 from djoser.serializers import UserCreatePasswordRetypeSerializer, UserSerializer, TokenCreateSerializer
 from rest_framework import serializers
 
+from profiles.models import Profile
 from validation.validate_password import validate_password_long, validate_password_include_symbols
 
 
 User = get_user_model()
 
 
+class CustomProfileSerializer(serializers.ModelSerializer):
+    comp_registered = serializers.BooleanField()
+    comp_is_startup = serializers.BooleanField()
+
+    class Meta:
+        model = Profile
+        fields = ("comp_name", "comp_registered", "comp_is_startup")
+
+
 class UserRegistrationSerializer(UserCreatePasswordRetypeSerializer):
-    person_email = serializers.EmailField(write_only=True)
+    company = CustomProfileSerializer(write_only=True)
+    person_email = serializers.EmailField(write_only=True, validators=[UniqueValidator(
+        queryset=User.objects.all(), message="Email is already registered")])
     password = serializers.CharField(
         style={"input_type": "password"}, write_only=True)
 
-    comp_registered = serializers.BooleanField()
-    comp_is_startup = serializers.BooleanField()
-
     class Meta(UserCreatePasswordRetypeSerializer.Meta):
         model = User
         fields = (
             "person_email",
             "password",
             "person_name",
             "person_surname",
-            "comp_name",
-            "comp_registered",
-            "comp_is_startup",
+            "company"
         )
 
-    def validate(self, attrs):
+    def validate(self, value):
+        custom_errors = {"password": [],
```

Comment:
```
custom_errors = defaultdict(list)
```

So you can:
1. Use constructions as `custom_errors["comp_status"].append(...`
2. .. and then at the end just `if custom_errors: raise ...`

Reply:
Fixed. Thanks.